code/modules/projectiles/guns/energy/experimental.dm

```diff
@@ -236,51 +241,267 @@
+			var/obj/item/stickybomb/SB = pick(fired)
+			spawn()
+				SB.detonate()
+			if(ismob(loc))
+				loc << "<span class='warning'>One of the stickybombs detonates to leave room for the next one.</span>"
+		fired += B
+		var/obj/item/projectile/stickybomb/SB = new()
+		SB.sticky = B
+		B.fired_from = src
+		B.loc = SB
+		in_chamber = SB
+		return 1
+	return 0
+
+
 /obj/item/stickybomb
-	name = "stickybomb"
-	desc = "."
+	name = "anti-personnel stickybomb"
+	desc = "Ammo for a stickybomb launcher."
 	icon = 'icons/obj/ammo.dmi'
-	icon_state = "s-casing"
+	icon_state = "stickybomb"
 	flags = FPRINT
 	siemens_coefficient = 1
 	slot_flags = SLOT_BELT
 	throwforce = 1
 	w_class = 1.0
+	var/obj/item/weapon/gun/stickybomb/fired_from = null
 	var/live = 0
-
-/obj/item/stickybomb/New()
+	var/atom/stuck_to = null
+	var/image/self_overlay = null
+	var/signal = 0
+
+/obj/item/stickybomb/Destroy()
+	if(fired_from)
+		fired_from.fired -= src
+		fired_from = null
+	stuck_to = null
+	self_overlay = null
 	..()
+
+/obj/item/stickybomb/update_icon()
+	icon_state = "[initial(icon_state)][live ? "-live" : ""]"
+	if(live)
+		desc = "It appears to be live."
+	else
+		desc = "Ammo for a stickybomb launcher."
+
+/obj/item/stickybomb/pickup(mob/user)
+	if(stuck_to)
+		user << "<span class='warning'>You reach for \the [src] stuck on \the [stuck_to] and start pulling.</span>"
+		if(do_after(user, src, 30))
+			user << "<span class='warning'>It came off!</span>"
+			unstick()
+			..()
+	else
+		..()
+
+/obj/item/stickybomb/proc/stick_to(var/atom/A as mob|obj|turf, var/side = null)
+	stuck_to = A
+	loc = A
 	pixel_x = rand(-10, 10)
 	pixel_y = rand(-10, 10)
+	playsound(A, 'sound/items/metal_impact.ogg', 30, 1)
+
+	if(isturf(A))
+		anchored = 1
+		switch(side)
+			if(NORTH)
+				pixel_y = 16
+			if(SOUTH)
+				pixel_y = -16
+			if(EAST)
+				pixel_x = 16
+			if(WEST)
+				pixel_x = -16
+		sleep(50)
+		if(stuck_to == A)
+			flick("stickybomb_flick",src)
+			live = 1
+			update_icon()
+			animate(src, alpha=50, time=300)
+
+	else if(isliving(A))
+		visible_message("<span class='warning'>\the [src] sticks itself on \the [A].</span>")
+		src.loc = A
+		self_overlay = new(icon,src,icon_state,10,dir)
+		self_overlay.pixel_x = pixel_x
+		self_overlay.pixel_y = pixel_y
+		A.overlays += self_overlay
+		sleep(50)
+		if(stuck_to == A)
+			live = 1
+			A.overlays -= self_overlay
+			self_overlay.icon_state = "stickybomb-live"
+			A.overlays += self_overlay
+
+/obj/item/stickybomb/proc/unstick(var/fall_to_floor = 1)
+	if(ismob(stuck_to))
+		stuck_to.overlays -= self_overlay
+		icon_state = self_overlay.icon_state
+		if(fall_to_floor)
+			src.loc = get_turf(src)
+	stuck_to = null
+	anchored = 0
+	alpha = 255
+	pixel_x = 0
+	pixel_y = 0
+
+/obj/item/stickybomb/proc/detonate()
+	icon_state = "stickybomb_flick"
+	if(!self_overlay)
+		self_overlay = new(icon,src,icon_state,13,dir)
+		overlays += self_overlay//a bit awkward but the sprite wouldn't properly animate otherwise
```

Comment:
well "stickybomb_flick" is a rapid flicker between to sprite, but for some reason when just changing the icon_state, the sprite gets stuck on the first frame, possibly because the atom gets deleted 3 ticks later or something I don't know. However I saw the sprite animate properly on stickies that were stuck to mobs (or rather their overlays). So By having this overlay on top of the not-animating sprite, It appears to flicker as intended.

I mean, sure in 3 game ticks most players won't even realize until they've blown up but it's the small things.